compiler/src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -108,12 +110,19 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
    */
   override def transformTemplate(template: Template)(using Context): Tree = {
     val cls = ctx.owner.asClass
-    appendOffsetDefs.get(cls) match {
-      case None => template
-      case Some(data) =>
-        data.defs.foreach(defin => defin.symbol.addAnnotation(Annotation(defn.ScalaStaticAnnot, defin.symbol.span)))
-        cpy.Template(template)(body = addInFront(data.defs, template.body))
-    }
+    if !selectImpl.useVarHandles then //ctx.settings.YlegacyLazyVals.value then
```

Comment:
```suggestion
    if !selectImpl.useVarHandles then
```
is this needed still?